python/lsst/ts/envsensors/socket_server.py

```diff
@@ -55,109 +57,106 @@ def __init__(self, port, simulation_mode=0):
 
         self.simulation_mode = simulation_mode
 
-        self._server = None
-        self._writer = None
-        self._reader = None
-        self._started = False
-        self.log = logging.getLogger(type(self).__name__)
+        self._server: tcpip.OneClientServer = None
+        self.read_loop_task: asyncio.Future = asyncio.Future()
+        self.log: logging.Logger = logging.getLogger(type(self).__name__)
 
-        self.command_handler = CommandHandler(
+        self.command_handler: CommandHandler = CommandHandler(
             callback=self.write, simulation_mode=self.simulation_mode
         )
 
-    async def start(self, keep_running=False):
-        """Start the TCP/IP server.
-
-        Start the command loop and make sure to keep running when instructed to
-        do so.
+    async def start(self, keep_running: bool = False) -> None:
+        """Start the TCP/IP server and the command loop.
 
         Parameters
         ----------
-        keep_running : bool
-            Used for command line testing and should generally be left to
-            False.
+        keep_running: `bool`
+            Indicates whether the server should keep running or not. Set to
+            True when started from the command line.
         """
         self.log.info("Start called")
-        self._started = True
-        self._server = await asyncio.start_server(
-            self.read_loop, host="127.0.0.1", port=self.port
+        self._server = tcpip.OneClientServer(
+            host=self.host,
+            port=self.port,
+            name=self.log.name,
+            log=self.log,
+            connect_callback=self.connected_callback,
         )
+        await self._server.start_task
 
         # Request the assigned port from the server so the code starting the
         # mock controller can use it to connect.
         if self.port == 0:
-            self.port = self._server.sockets[0].getsockname()[1]
+            self.port = self._server.port
```

Comment:
If you subclass from OneClientServer then you don't need this.

Or if you continue to construct the OneClientServer in this class's constructor, then I suggest making this a property instead of copying the value. That avoids duplicating data, which can get out of sync.

Reply:
I changed to subclassing so I removed it.